app/packs/src/models/ResearchPlan.js

```diff
@@ -267,4 +268,37 @@ export default class ResearchPlan extends Element {
     return this.attachments
       .filter((attachment) => attachment.is_deleted === true && !attachment.is_new);
   }
+
+  buildCopy(params = {}) {
+    const copy = super.buildCopy();
+    Object.assign(copy, params);
+    copy.attachments = this.attachments;
+    copy.container = Container.init();
+    copy.is_new = true;
+    copy.is_copy = true;
+    copy.can_update = true;
+    copy.can_copy = true;
+
+    return copy;
+  }
+
+  static copyFromResearchPlanAndCollectionId(research_plan, collection_id) {
```

Comment:
i think here are some more elements of the RP missing: metadata, references, wellplates, ...

And of course the generic segments